NBitcoin/JsonConverters/CoinJsonConverter.cs

```diff
@@ -62,7 +62,7 @@ public ICoin ToCoin(string path)
 
 				
 				var coin = RedeemScript == null ? new Coin(new OutPoint(TransactionId, index), new TxOut(Value, ScriptPubKey)) : new ScriptCoin(new OutPoint(TransactionId, index), new TxOut(Value, ScriptPubKey), RedeemScript);
-				if (AssetId != null)
+				if (!(AssetId is null))
```

Comment:
In C# 8.0 there is this `if (variable is { })` which reads as if variable is something or is object (not null)
OR this `if (variable is object)`
The latter is maybe more readable but my preference is the former as there are more things you can do with it,. Check this video.
https://www.youtube.com/watch?v=TJiLhRPgyq4&t=622s
https://intellitect.com/check-for-null-not-null/

Reply:
WTF how could I be unaware of `if(variable is { })` ?!?

Reply:
`variable is { }` is what `variable is var` should have been from day one.

Reply:
@NicolasDorier Look at the youtube video @yahiheb linked. There are crazy things to do with `is { }`, we decided on using that from yesterday.

Reply:
I asked them about it on https://twitter.com/NicolasDorier/status/1196449885538549761?s=20

Apparently `{}` is not a keyword but part of the "property based pattern matching". Which I completely ignored when I read as I did not see the purpose.

Reply:
@nopara73 @NicolasDorier https://devblogs.microsoft.com/dotnet/welcome-to-c-9-0/#logical-patterns
With C# 9.0 you can simply write `if (AssetId is not null)`

Reply:
@yahiheb oh my god they implemented it! I thought there was some tough discussion drama around it.

Reply:
![image](https://user-images.githubusercontent.com/9156103/83111722-51d2ea80-a0c5-11ea-8025-f14b7049a8c7.png)